src/integrationTests/EdgeCases.test.js

```diff
@@ -0,0 +1,220 @@
+import * as _ from 'lodash';
+
+['es6Remx', 'legacyRemx'].forEach((version) => {
+  describe(`EdgeCases (${version})`, () => {
+    let remx, mobx;
+    let state, setters, getters, anotherState;
+
+    beforeEach(() => {
+      mobx = require('mobx');
+      remx = require(`../${version}/remx`);
+
+      anotherState = remx.state({
+        count: 0
+      });
+
+      state = remx.state({
+        obj: {
+          foo: 'bar'
+        }
+      });
+      setters = remx.setters({
+        modify() {
+          state.anotherState = anotherState;
+          anotherState.count++;
+        },
+        add(o) {
+          state.added = { o };
+        }
+      });
+      getters = remx.getters({
+        getCount() {
+          return state.anotherState ? state.anotherState.count : -1;
+        }
+      });
+    });
+
+    afterEach(() => {
+      delete global.__mobxInstanceCount; // prevent warnings
+    });
+
+    it('support setting partial remx state inside another state', () => {
+      const state1 = remx.state({ obj1: { arr1: [] } });
+      const state2 = remx.state({});
+
+      const getter1 = remx.getters({ getObj1: () => state1.obj1 });
+      const setter2 = remx.setters({ set: () => state2.someProp = getter1.getObj1() });
+
+      let callCount = 0;
+      const stop = mobx.autorun(() => {
+        JSON.stringify(getter1.getObj1());
+        callCount++;
+      });
+      expect(callCount).toEqual(1);
+      setter2.set();
+      const expectedCallCounts = { es6Remx: 1, legacyRemx: 2 };
+      expect(callCount).toEqual(expectedCallCounts[version]);
+      stop();
+      expect(state2.someProp).toEqual({ arr1: [] });
+    });
+
+    it('supports partial state from another state with arrays', () => {
+      const key1 = { name: 'key1' };
+      const key2 = { name: 'key2' };
+      const state1 = remx.state({ obj1: { arr1: [key1, key2] } });
+      const state2 = remx.state({});
+
+      const getter1 = remx.getters({ getArr: () => state1.obj1.arr1 });
+      const setter2 = remx.setters({ set: (k) => state2.someProp = k });
+
+      let callCount = 0;
+      const stop = mobx.autorun(() => {
+        JSON.stringify(getter1.getArr());
+        callCount++;
+      });
+      expect(callCount).toEqual(1);
+      setter2.set(getter1.getArr()[0]);
+      const expectedCallCounts = { es6Remx: 1, legacyRemx: 2 };
```

Comment:
Note that behaviour is different
Would be great to make es6Remx work like legacyRemx in this case